### PR TITLE
TFAC-294 Setting Target=_top for External Links

### PR DESCRIPTION
**Pull request by jedtan**
Modify the `Link` component:
* If the link URL is not on the current domain or the path does not start with `/newtab/`, use `target="_top"` by default
* If the link URL is on the current domain and the path starts with `/newtab/`, don’t set a default target

QA'd manually:
* the links conditionally add the target as expected
* when the app is in an iframe, external links navigate the top frame
* when the app is in an iframe, internal (same-app) links navigate in the nested frame

### CODE REVIEW

File: src/components/Link.js [hunk 1]
```diff
@@ -15,10 +16,23 @@ const useStyles = makeStyles(() => ({
 const Link = (props) => {
   const { children, className, to } = props
   const classes = useStyles()
+  const [destInternal, setDestInternal] = useState(true)
+
+  useEffect(() => {
+    if (isURLForDifferentApp(to)) {
+      // Set that we can render the Firebase auth UI.
```
Comment (kmjennison) [inline src/components/Link.js:23]: Remove unrelated comment.

File: src/utils/navigationUtils.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  !!(url.startsWith('http://') || url.startsWith('https://'))
+
+/**
+ * Determine whether a URL is for a different app from the current
+ * URL. In other words, navigating to the new URL cannot simply
+ * navigate within a single-page app.
+ * @param {String} newURL - The URL to test
+ * @return {Boolean} Whether the new URL is for another app
+ */
+export const isURLForDifferentApp = (newURL) => {
```
Comment (kmjennison) [inline src/utils/navigationUtils.js:12]: Make some changes and simplify this:
* Don't consider absolute URLs a different app
* Remove any logic about search
* Remove any logic about homepage

File: src/utils/testHelpers/testUtils.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* globals window */
+
```
Comment (kmjennison) [inline src/utils/testHelpers/testUtils.js:2]: Rename file to be explicit: `setWindowLocation.js`

**Review by kmjennison — COMMENTED** [review]
Nice, thanks! I'm going to jump in now to wrap this up.

**Review by amaliwanag2 — APPROVED** [review]
LGTM :cheese:

**Pull request merged**